### DISCUSSION TITLE
windscribe.com: nuisance

### DISCUSSION THREAD

**Discussion by MasterKia** [uBlockOrigin/uAssets#13773] [General]
### Prerequisites

- [X] This is not a [support issue or a question](https://old.reddit.com/r/uBlockOrigin/)
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uAssets/issues) to avoid opening a duplicate issue
- [X] I am running the latest version of uBlock Origin

### I tried to reproduce the issue when...

- [X] uBO is the only extension
- [X] uBO with default lists/settings
- [X] using a new, unmodified browser profile

### URL(s) where the issue occurs

`https://windscribe.com/signup`

### Describe the issue

If I'm using Windscribe, I can't create another account (to give to someone else for example, because VPN site is blocked here so people without VPN can't access the site to create an account).

### Screenshot(s)

`You already have a Windscribe account. Creating multiple accounts is prohibited.`
https://user-images.githubusercontent.com/17685483/174468730-dd0c7ca2-10a8-4777-a982-87be305feccc.png

### uBlock Origin version

1.43.0

### Browser name and version

Chrome

### Settings

-

### Notes

Client-side check
`windscribe.com##+js(set, ha_int, 0)` doesn't work.

```js
signup_click = false;
$("#signupform").submit(function(e) {
            var buffer = (Date.now() / 1000) - (86400 * 2);
            var ha_int = parseInt(ha);
            if (ha_int >= buffer) {
                $('.content_message').text("You already have a Windscribe account. Creating multiple accounts is prohibited.").show();
                return false;
            }
```

**Comment by ghost** [comment]
`ha_int` is local in this function. uBO overrides the top global variable. Aborting it will break the "submit" function. `ha` is from local storage. This is unlikely, but maybe `set` on `window.localStorage.ha` will work?

**Reply by MasterKia** [reply]
> uBO overrides the top global variable [and doesn't override local variables]

Is this a bug or is it by design?

Because sometimes it's handy to override _local_ variables.
For example in https://github.com/uBlockOrigin/uAssets/issues/13658#issuecomment-1152868003, `##+js(set, input, {})` didn't work to override a variable inside function to defuse tracking.

**Reply by ghost** [reply]
It's not possible to access them from outside (from any other script or even function).

**Reply by ghost** [reply]
For objects, it is sometimes possible to override prototype. Int is just a simple variable.

**Reply by MasterKia** [reply]
> but maybe `set` on `window.localStorage.ha` will work?

`##+js(set, window.localStorage.ha, 0)`? tried, didn't work.

`##+js(aopr, window.localStorage)` does kinda work, but after clicking on "Create account":
![image](https://user-images.githubusercontent.com/17685483/174473668-4a4c2693-ee34-45a8-8594-6cd1a7925522.png)


Is the check now suddenly server-side? I don't think so.

**Reply by MasterKia** [reply]
What about deleting that part from site code?
Like how `##noscript:remove()` can remove "noscript" tag, or using HTML filtering (but Firefox only unfortunately).

**Reply by stephenhawk8054** [reply]
You can only remove a whole script under a tag, cannot remove part of a script.

**Reply by MasterKia** [reply]
How they are able to detect the VPN being on without checking against an IP list is beyond me.

**Reply by mapx-** [reply]
They are checking it (the ip) on server side (probably a cloudflare feature - that's why that generic `Access denied` cloudflare page)
